test/parallel/test-http-upgrade-advertise.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const tests = [
+  { headers: {}, expected: 'regular' },
+  { headers: { upgrade: 'h2c' }, expected: 'regular' },
+  { headers: { connection: 'upgrade' }, expected: 'regular' },
+  { headers: { connection: 'upgrade', upgrade: 'h2c' }, expected: 'upgrade' }
+];
+
+function fire() {
+  if (tests.length === 0)
+    return server.close();
+
+  const test = tests.shift();
+
+  var once = false;
+
+  const done = common.mustCall(function done(result) {
+    assert(!once);
```

Comment:
Not strictly necessary, `common.mustCall()` will squeal if the callback is called more than once (at script exit though.)